Entities/ADoor.cpp

```diff
@@ -1,921 +1,598 @@
+				if (foundActor->GetTeam() != m_Team) {
+					CloseDoor();
+					break;
+				} else if (foundActor->GetTeam() == m_Team) {					
+					OpenDoor();
+				}
+			}
+		}
+		// If no sensor input, return to default state if it's time. This won't override any actions that the sensors started
+		if (!anySensorInput && m_ResetDefaultTimer.IsPastSimMS(m_ResetDefaultDelay)) {
+			if (m_ClosedByDefault && m_DoorState == OPEN) {
+				CloseDoor();
+			} else if (!m_ClosedByDefault && m_DoorState == CLOSED) {
+				OpenDoor();
+			}
+		}
+		m_SensorTimer.Reset();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::UpdateDoorAttachable() {
+		m_Door->SetHFlipped(m_HFlipped);
+		switch (m_DoorState) {
+			case OPEN:
+				m_Door->SetJointPos(m_Pos + m_OpenOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+				m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (m_OpenAngle * GetFlipFactor()));
+				break;
+			case OPENING:
+				if (!m_DoorMoveSound.IsBeingPlayed()) { m_DoorMoveSound.Play(m_Pos); }
+
+				if (m_DoorMoveTimer.IsPastSimMS(m_DoorMoveTime)) {
+					m_ResetDefaultTimer.Reset();
+
+					m_Door->SetJointPos(m_Pos + m_OpenOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+					m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (m_OpenAngle * GetFlipFactor()));
+
+					// Draw the material representation of the door to the terrain in the open state
+					if (m_DrawMaterialLayerWhenOpen) { DrawDoorMaterial(); }
+
+					m_DoorMoveSound.Stop();
+					m_DoorMoveEndSound.Play(m_Pos);
+
+					m_DoorState = OPEN;
+				} else {
+					// Find interpolated positions and angles for opening
+					Vector openingOffset(LERP(0, m_DoorMoveTime, m_ClosedOffset.m_X, m_OpenOffset.m_X, m_DoorMoveTimer.GetElapsedSimTimeMS()), LERP(0, m_DoorMoveTime, m_ClosedOffset.m_Y, m_OpenOffset.m_Y, m_DoorMoveTimer.GetElapsedSimTimeMS()));
+					// TODO: Make this work across rotation 0
+					float openingAngle = LERP(0, m_DoorMoveTime, m_ClosedAngle, m_OpenAngle, m_DoorMoveTimer.GetElapsedSimTimeMS());
+
+					m_Door->SetJointPos(m_Pos + openingOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+					m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (openingAngle * GetFlipFactor()));
+
+					// Do the deep check to clear away any terrain debris when the door is moving but only after a little while so it doesn't take a chunk out of the ground before being snapped into place
+					if (m_DoorMoveTimer.IsPastSimMS(50)) { m_Door->DeepCheck(true); }
+				}
+				break;
+			case CLOSED:
+				m_Door->SetJointPos(m_Pos + m_ClosedOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+				m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (m_ClosedAngle * GetFlipFactor()));
+				break;
+			case CLOSING:
+				if (!m_DoorMoveSound.IsBeingPlayed()) { m_DoorMoveSound.Play(m_Pos); }
+
+				if (m_DoorMoveTimer.IsPastSimMS(m_DoorMoveTime)) {
```

Comment:
Same stuff about comments and so on applies as in OPENING case

Honestly it might be nice to pull all this logic out into some private helper UpdateDoorAttachableOpeningAndClosing method and reduce code duplication + trim down this method. Dunno if that's difficult though.